collections/_posts/2022-06-22-Fed2-FeatureAlignedFederatedLearning.md

```diff
@@ -0,0 +1,63 @@
+---
+layout: review
+title: "Fed2: Feature-Aligned Federated Learning"
+tags: Federated classification
+author: "Matthis Manthe"
+cite:
+    authors: "Yu, Fuxun, Weishan Zhang, Zhuwei Qin, Zirui Xu, Di Wang, Chenchen Liu, Zhi Tian, Xiang Chen"
+    title:   "Fed2: Feature-Aligned Federated Learning"
+    venue:   "Proceedings of the 27th ACM SIGKDD Conference on Knowledge Discovery & Data Mining, KDD ’21."
+pdf: "https://arxiv.org/abs/2111.14248"
+---
+
+# The notion of permutation invariance
+
+* Given a Fully Connected Neural Net (FCNN) with two consecutive layers, one can swap the positions of two neurons in a layer and their associated weights entering and leaving the neurons without altering the outputs of the network. Formally, the FCNN $$F$$ with weights $$w_0$$ and $$w_1$$ given input X is invariant to any permutation matrix:
+
+$$F(X) = w_1w_0X = (w_1\Pi)(\Pi^Tw_0)X, \Pi \text{ a permutation matrix}$$ 
+
+* The same happens to Convolutional Networks and their filters.
+* It implies a complex problem in Model Fusion and Federated Learning: **How to be sure that the neurons combined by a parameters average encode the same learned features ?**.
```

Comment:
I find the formulation "combined by a parameters average encode" is weirdly constructed and leads to confusing interpretations, since "average" can be a verb (that's how I initially read it). So, if I understood what you meant correctly, I would suggest a phrasing that is IMO easier to read:
```suggestion
* It implies a complex problem in Model Fusion and Federated Learning: **How to be sure that the neurons combined by averaging parameters encode the same learned features ?**.
```

Reply:
Yes that's better indeed.